spec/QueryTools.spec.js

```diff
@@ -587,4 +587,49 @@ describe('matchesQuery', function() {
     q.notContainedIn('profile', ['abc', 'def', 'ghi']);
     expect(matchesQuery(message, q)).toBe(false);
   });
+
+  it('matches on Date', () => {
+    // given
+    const now = new Date();
+    const obj = {
+      id: new Id('Person', '01'),
+      dateObject: now,
+      dateJSON: {
+        __type: 'Date',
+        iso: now.toISOString(),
+      },
+    };
+
+    // when, then: Equal
+    let q = new Parse.Query('Person');
+    q.equalTo('dateObject', now);
+    q.equalTo('dateJSON', now);
+    expect(matchesQuery(Object.assign({}, obj), q)).toBe(true);
+
+    // when, then: lessThan
+    const future = Date(now.getTime() + 1000);
```

Comment:
LiveQuery does support date, I just tried it.

Looks you aren't using `new Date` which returns an object that gets decoded.

Reply:
I found that the Date filed of the object inside the matchesKeyConstraints function is represented by Date JSON {__type: "Date", iso: "xxxx-xx-xxTyy:yy:yy.yyyZ"}, not Date object.
Thus it should decode it to Date JSON to compare with constraint.

P.S. I test on parse-server with iOS Client.

Reply:
I realized that after I posted the comment 👍